.github/workflows/esrupgrade.yml

```diff
@@ -0,0 +1,92 @@
+name: ESR Upgrade
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - cloud
+      - release-*
```

Comment:
Yeah, this was copied from the other actions in the server. I don't have any preference regarding this, as long as it runs when we modify the files.

Is there a way to run the workflow manually?

Reply:
Answering myself: https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

However, running it manually doesn't seem to provide a good dev experience. We could have the same triggers we have now, but using the `paths` filter https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Reply:
My concern is why we should run this on every PR? What's going to change that often? We have *not so strict* rule to not touching older migrations. That means this will always run `v5.37` -> `v7.8` scripts. Isn't it?

Reply:
Forgot to answer this before: we can simply run this on master (or on the release and cloud branches only), and on pull requests that modify files inside scripts/esrupgrades or .github/workflows/esr*. So something like (untested):
```yml
on:
  pull_request:
  push:
    branches:
      - cloud
      - release-*
    paths:
      - 'scripts/esrupgrades/*'
      - '.github/workflows/esr*'
```

Reply:
Makes sense to me. Thanks!

Reply:
Done in https://github.com/mattermost/mattermost-server/pull/22448/commits/01a47e96fbee8362860683ef0cfe28cb9d13acb9